Tests/scripts/slack_notifier.py

```diff
@@ -111,56 +114,137 @@ def options_handler():
-    color = 'good' if not unittests_fields else 'danger'
-    if not unittests_fields:
-        title = 'Content Nightly Unit Tests - Success' if not is_sdk_build else 'SDK Nightly Unit Tests - Success'
+def get_failed_unit_tests_attachment(build_url: str, is_sdk_build: bool = False) -> List[Dict]:
+    """
+    Returns the failed unit tests attachment to be reported in Slack.
+    Args:
+        build_url (str): Build URL of the given nightly.
+        is_sdk_build (bool): Whether build is SDK nightly or content nightly.
+
+    Returns:
+        (List[Dict]) Dict wrapped inside a list containing failed unit tests attachment.
+    """
+    if artifact_data := get_artifact_data('failed_lint_report.txt'):
+        artifact_data = artifact_data.split('\n')
+        unittests_fields: Optional[List[Dict]] = get_entities_fields(f'Failed Unittests - ({len(artifact_data)})',
+                                                                     artifact_data)
     else:
-        title = 'Content Nightly Unit Tests - Failure' if not is_sdk_build else 'SDK Nightly Unit Tests - Failure'
-    content_team_attachment = [{
+        unittests_fields = []
+    color: str = 'good' if not unittests_fields else 'danger'
+    build_type: str = 'SDK' if is_sdk_build else 'Content'
+    status = 'Success' if not unittests_fields else 'Failure'
+    title: str = f'{build_type} Nightly Unit Tests - {status}'
+    return [{
         'fallback': title,
         'color': color,
         'title': title,
         'title_link': build_url,
         'fields': unittests_fields
     }]
-    return content_team_attachment
+
+
+def get_coverage_color(coverage_percent: float) -> str:
+    """
+    Returns color to represent coverage percent.
+    Args:
+        coverage_percent (float): Coverage percent.
+
+    Returns:
+        (str): Representing the color
+    """
+    if coverage_percent <= 50.0:
+        return 'danger'
+    elif coverage_percent < 60.0:
+        return 'warning'
+    return 'good'
+
+
+def get_coverage_attachment(build_url: str) -> Optional[Dict]:
+    """
+    Returns content coverage report attachment.
+    Args:
+        build_url (str): Build URL of the nightly.
+
+    Returns:
+        (Dict): Attachment of the coverage if coverage report exists.
+    """
+    xml_coverage_data: str = get_artifact_data('coverage_report/coverage.xml')
+    if not xml_coverage_data:
+        return None
+    coverage_dict_data: OrderedDict = xmltodict.parse(xml_coverage_data)
+    coverage_percent: float = float(coverage_dict_data.get('coverage', dict()).get('@line-rate')) * 100.0
+    return {
+        'fallback': f'Coverage Report Content: {coverage_percent:.2f}% Total Coverage',
+        'color': get_coverage_color(coverage_percent),
+        'title': f'Coverage Report Content: {coverage_percent:.2f}% Total Coverage',
+        'title_link': build_url,
+        'fields': []
```

Comment:
Do we need this line if the the value is an empty list?

Reply:
Not sure to be honest. But it does work this way as expected. I can perform checks without it if you think it is better without.

Reply:
Not a big deal. Can leave it as it is.